### ISSUE TITLE
asdoc JDK 11 "-Xbootclasspath/p is no longer a supported option"

### ISSUE THREAD

**Issue by pol2095** [airsdk/Adobe-Runtime-Support#1910]
## Problem Description

Impossible to generate asdoc using JDK 11

* AIR SDK 33.1.1.795 on Windows
* Java(TM) SE Runtime Environment 18.9 (build 11.0.13+10-LTS-370)

## Steps to Reproduce

when I use asdoc using JDK 11, this error occur
```
-Xbootclasspath/p is no longer a supported option.
Error: Could not create the Java Virtual Machine.
```

## Known Workarounds

use JDK 8

**Comment by ajwfrost** [comment]
This is a curious one. The argument there is trying to provide the class path for "xalan.jar". In Flex 4.5 I can see that this is in the "lib" folder but with Flex 4.16 it was moved into lib/external. In AIR ('new compiler') variant it's not actually present!

The asdoc script for Flex 4.16 is referencing it in "lib" still .. which is not where it lies. The asdoc script for AIR is referencing it in the "asdoc" top-level folder, but it's not there.

In other words, this library isn't actually being loaded and when I take out that command, it's not causing any problems in our asdoc generation...!

We can update the scripts to remove this reference, for the AIR SDK; you'll have to manually do this if you're using a Flex SDK with the AIR overlay....

thanks

**Comment by pol2095** [comment]
Thanks, how to remove the reference manually in the SDK AIR ?

**Comment by pol2095** [comment]
I don't understand how it work using JDK 8 ?

**Comment by ajwfrost** [comment]
It would work in Java 8 because the error you're seeing is about the arguments to the java.exe process. The "-Xbootclasspath" arguments are no longer supported in Java 11, so just the fact that they're there is enough to cause it to complain.

The actual library/path being provided in that argument doesn't exist, but that doesn't cause any error (when using Java 8) as it's only something that would be searched if it was needed...

I actually noticed that the equivalent line is not present in the macOS version of the asdoc script...

macOS/Linux (asdoc):
```
java $VMARGS $SETUP_SH_VMARGS -Dflexcompiler="$FALCON_HOME" -Dflexlib="$FLEX_HOME/frameworks" -jar "${FALCON_HOME}/lib/legacy/asdoc.jar" "-compiler.fonts.local-fonts-snapshot=" "$@"
```
vs Windows (asdoc.bat):
```
java -Dsun.io.useCanonCaches=false -Xms32m -Xmx512m -Dflexcompiler="%FALCON_HOME%" -Dflexlib="%FLEX_HOME%/frameworks" -Dflex.compiler.theme= -Xbootclasspath/p:"%FLEX_HOME%/asdoc/xalan.jar" -classpath "%FLEX_HOME%/lib/legacy/asdoc.jar" flex2.tools.ASDoc -compiler.fonts.local-fonts-snapshot= %*
```
which should instead be:
```
java -Dsun.io.useCanonCaches=false -Xms32m -Xmx512m -Dflexcompiler="%FALCON_HOME%" -Dflexlib="%FLEX_HOME%/frameworks" -Dflex.compiler.theme= -classpath "%FLEX_HOME%/lib/legacy/asdoc.jar" flex2.tools.ASDoc -compiler.fonts.local-fonts-snapshot= %*
```

thanks

**Comment by pol2095** [comment]
Thanks

**Comment by pol2095** [comment]
The reference `-Xbootclasspath/p:"%FLEX_HOME%/asdoc/xalan.jar"` is still present in `<AIR_SDK>\bin\asdoc.bat` and causes the error using Java 11 
```
-Xbootclasspath/p is no longer a supported option.
Error: Could not create the Java Virtual Machine.
```

**Comment by ajwfrost** [comment]
Wow .. found that we'd made the change in 33.1 but then hadn't pulled that across into any of the 50.x branches!
Have just pulled it into 50.2 so it will be in the next SDK release. Thanks for spotting it!

**Comment by pol2095** [comment]
Thanks

**Comment by pol2095** [comment]
Thanks
